### PR TITLE
Fix toHaveTextContent to be case-sensitive

### LINKED ISSUE #71: Should .toHaveTextContent be case-sensitive?
I don't know if this is a bug or it is a feature, but I expected that `.toHaveTextContent` was case-sensitive. I could address that if it is a bug.

* `jest-dom` version: 2.1.1

### What you did:

```html
<div className="App">
  <h1>Hello CodeSandbox</h1>
  <h2>Start editing to see some magic happen!</h2>
</div>
```
I expect the expect below would fail, but it passes.
```js
expect(subject.container.querySelector("h1")).toHaveTextContent("hello codesandbox");
```

### PR DESCRIPTION
**Pull request by lourenci**
**What**:
Change `.toHaveTextContent` to be case-sensitive. It closes #71.

**Why**:
This is the expected usage. If the user wants insensitivity, he should do through `Regexp`.

**How**:
Is was removed the `.toLowerCase` from the `match` utils function.

**Checklist**:

* [ ] Documentation
* [x] Tests
* [x] Updated Type Definitions
* [x] Ready to be merged 
* [x] Added myself to contributors table

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ function matches(textToMatch, matcher) {
   if (matcher instanceof RegExp) {
     return matcher.test(textToMatch)
   } else {
-    return textToMatch.toLowerCase().includes(String(matcher).toLowerCase())
+    return textToMatch.includes(matcher)
```
Comment (gnapse) [inline src/utils.js]: I would keep `matcher` wrapped in `String()`. This allows to pass non string stuff (such as numbers) to `.toHaveTextContent`, and removing it as part of this change adds more surface for tests to break for issues not related with what you're trying to fix (case sensitiveness).

```suggestion
    return textToMatch.includes(String(matcher))
```

Comment (lourenci) [reply src/utils.js]: It makes sense. I'll do it.

**Comment by lourenci** [comment]
Yes, it's a breaking change. Some tests written believing in insensitivity will break.

**Pull request merged**